snowplow-common-internal/kinesis/src/it/scala/com/snowplowanalytics/snowplow/sources/kinesis/Resources.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2023-present Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Snowplow Community License Version 1.0,
+ * and you may not use this file except in compliance with the Snowplow Community License Version 1.0.
+ * You may obtain a copy of the Snowplow Community License Version 1.0 at https://docs.snowplow.io/community-license-1.0
+ */
+package com.snowplowanalytics.snowplow.sources.kinesis
+
+import cats.effect.{IO, Ref}
+import com.snowplowanalytics.snowplow.sources.{EventProcessor, TokenedEvents}
+import org.slf4j.LoggerFactory
+import org.testcontainers.containers.localstack.LocalStackContainer
+import org.testcontainers.containers.output.Slf4jLogConsumer
+import org.testcontainers.containers.wait.strategy.Wait
+import org.testcontainers.utility.DockerImageName
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.kinesis.KinesisAsyncClient
+
+import java.net.URI
+import java.nio.charset.StandardCharsets
+
+object Resources {
+
+  def prepareLocalstackContainer(region: Region, kinesisInitializeStreams: String): LocalStackContainer = {
```

Comment:
```suggestion
  def resource(region: Region, kinesisInitializeStreams: String): Resource[LocalStackContainer] = {
```
IMHO this should create the localstack resource directly, so call the `start` and the `stop`.

I also think that this file should only contain localstack stuff, so it can probably be renamed `Localstack` and this function `resource`